idseq_dag/engine/pipeline_step.py

```diff
@@ -96,6 +97,28 @@ def uploading_results(self):
             idseq_dag.util.s3.upload_folder_with_retries(f, self.s3_path(f))
         self.status = StepStatus.UPLOADED
 
+    def update_stage_status_json(self):
```

Comment:
There is a problem here with concurrency. Multiple steps of the pipeline are run simultaneously, so we need to either block the file (or better the read/write operations) during this function or we need to revise the logic to use separate files.
If you see below, as we read in and write in separate blocks, another thread can write to this file in between. In this case, the write update could erase info added by other process.
Check with @davidrissato on the best way to do that, given what has been used in dag before.

Reply:
I added a lock that is shared between all the steps in the same stage, which the steps must acquire in order to access the file